### ISSUE TITLE
Thunder Scans: Premium/Paid Chapters Auto detect

### ISSUE THREAD

**Issue by Batlh** [keiyoushi/tachiyomi-extensions-source#5908]
### Source name

Thunder Scans

### Source language

English

### Describe your suggested feature

Currently, early access chapter do not appear in the app as a paid/premium chapter.

If there was a toggle in settings so they would show up as a paid/premium chapter or a way it could auto detect that these are paid/premium chapters, and don't list them till they are free to read, would be great.
The other option would be to have us logging in with our accounts to access the Premium/paid chapters for those who already have a premium account.

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] If this is an issue with the app itself, I should be opening an issue in the [app repository](https://github.com/mihonapp/mihon/issues/new/choose).
- [X] I have updated the app to version **[0.15.3](https://github.com/mihonapp/mihon/releases/latest)**.
- [X] I will fill out all of the requested information in this form.

**Comment by prem-k-r** [comment]
Also, Manga Galaxy and Vortex Scans please

**Comment by Smol-Ame** [comment]
@prem-k-r make your own feature request about rather than commenting in a feature request for an unrelated source/request.

**Comment by mrtear** [comment]
I just wanted to add that the source doesn’t show all the paid chapters—it only displays the latest locked/paid chapter (as you can see in the GIF below). It’s not just one; all of them(paid ones) are like this.

![error.gif](https://github.com/user-attachments/assets/ef304033-cfa0-48b1-89f1-98b55b9430d8)

**Comment by Batlh** [comment]
> I just wanted to add that the source doesn’t show all the paid chapters—it only displays the latest locked/paid chapter (as you can see in the GIF below). It’s not just one; all of them(paid ones) are like this.
> 
> ![error.gif](https://github.com/user-attachments/assets/ef304033-cfa0-48b1-89f1-98b55b9430d8)
> 
> 

This is a new behaviour for the source.... i also encountered it today, so I can confirm that it shows only the newest paid chapter and withholds the older paid chapters...

Maybe we should include this as a bug report in an extra issue?

**Comment by mrtear** [comment]
> Maybe we should include this as a bug report in an extra issue?

Hmm maybe they'll see this and address it along with this request.

**Comment by mneko** [comment]
Hopefully this is next on the list now that this issue got solved for some other sources in #6155

**Comment by AgentX2006** [comment]
Since the bug is not listed here I'll mention it. This also causes another issue in Tachiyomi when using Auto download while reading. When it goes to download a paid chapter it buggs the download queue and all downloads are stuck waiting on it until you manually clear the paid chapter that keeps failing from the queue.

**Comment by vetleledaal** [comment]
@AgentX2006 this is the result of having chapters that can't be downloaded in the list. The issue mostly stems from the extension API, meaning it can't be fixed on either side if you want to show unavailable chapters.

[There is a proposed API change](https://github.com/AntsyLich/mhx-api-rfc/blob/main/RFC.md#chapter), where `lockedStatus` can be used to solve this exact issue. This is however not expected to be implemented any time soon.

**Comment by AgentX2006** [comment]
I'm not sure if it's somehow different with this source but Asura Scans announced paid chapters 4 days ago and that extension already has this feature implemented. Is there a reason it's taking so long for Thunder Scans? We've been waiting over a month now.

**Comment by Smol-Ame** [comment]
@AgentX2006 There's over 480~ issues in the repo, a lot of them are older than a month. All the work is done by contributors volunteering their time & efforts, and not everything is cookie cutter ready/easy to fix/implement.

There's already a proposed/work-in-progress pull request (#6598) to address this issue, but it's under discussion right now. Don't unnecessarily comment if you have nothing useful to contribute, since it emails everyone involved in the issue & pings the Discord server.
